File: pisio-backend/incident_report/src/main/java/com/example/pisio/incident_report/models/IncidentRequest.java
package com.example.pisio.incident_report.models;

import com.example.pisio.incident_report.models.entities.IncidentSubtypeEntity;
import com.example.pisio.incident_report.models.entities.LocationEntity;
import com.example.pisio.incident_report.models.entities.ModeratorEntity;
import lombok.Data;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Setter
public class IncidentRequest {


    private String description;
    private String photoLink;
    private LocalDateTime dateOfReport;
    //private LocalDate dateOfApproval;
    private Boolean approved;


   // private Integer subtypeId;

    private BigDecimal longitude;

    private BigDecimal latitude;


    private IncidentSubtype incidentSubtype;

    private Location location;


}

File: pisio-frontend/main/src/app/models/incident_request.ts
import { IncidentSubtype } from "./incident_subtype";
import { IncidentType } from "./incident_type";
import {Location} from "./location";


export interface IncidentRequest{

    id: number | null; 
    description : string;  
    incidentSubtype : IncidentSubtype; 
    photoLink: string | null; 
    longitude : number; 
    latitude: number;
    approved: number;
    dateOfReport: string;

}